src/controller/order-controller.ts

```diff
@@ -1,97 +1,168 @@
-          } as PurchaseItem;
-        }),
+        items:
+          itemsArray /* Line item detail can be seen in the PayPal Checkout by clicking the amount in the upper-right, and is stored in the transaction record */,
+
+        /* invoice_id: invoiceId,  /* Your own unique order #, will be indexed and stored as part of the transaction record and searchable in paypal.com account
+                                      (Must be unique, never used for an already *successful* transaction in the receiving account;
+                                      payment attempts with the invoice_id of a previously successful transaction are blocked to prevent accidental repeat payment for same thing) */
+
+        // custom_id: "any-arbitrary-metadata-value",  /* Not indexed nor searchable, but value will be returned in all API or webhook responses and visible in the transaction record of *receiving* PayPal account */
       },
     ],
-  };
 
-  const data = await createOrder(accessToken, orderPayload);
-  reply.send(data);
+    /*
+    payment_source: {
+      paypal: {
+        experience_context: {
+          // If there are no tangible items to be shipped as part of the transaction, you can specify NO_SHIPPING to not collect the payer's shipping address
+          // shipping_preference: "NO_SHIPPING",
+
+          // If after the payer's approval at PayPal you are *not* going to capture the Order ID immediately, but rather show final review step(s),
+          // set the user_action to CONTINUE (default with JS SDK approval is PAY_NOW). This is a cosmetic change to the wording of the last button at PayPal,
+          // so that it reads according to the action your integration performs on return.
+          // In summary: if CONTINUE is set here, your onApprove callback should proceed to your own review step that *requires a user action* before capture.
+          // user_action: "CONTINUE",
+        },
+      },
+    },
+    */
+  }; //as CreateOrderRequestBody; //cast needed for payment_source.paypal with paypal-js@5.1.4
+
+  const { data, httpStatus } = await createOrder(orderPayload);
+
+  reply.code(httpStatus).send(data);
 }
 
 async function captureOrderHandler(
   request: FastifyRequest,
   reply: FastifyReply
 ) {
   const { orderID } = request.body as { orderID: string };
-  const { access_token: accessToken } = await getAuthToken();
 
-  const data = await captureOrder(accessToken, orderID);
-  reply.send(data);
+  const { data, httpStatus } = await captureOrder(orderID);
+
+  const transaction =
+    data?.purchase_units?.[0]?.payments?.captures?.[0] ||
+    data?.purchase_units?.[0]?.payments?.authorizations?.[0];
+
+  if (!transaction?.id || transaction.status === "DECLINED") {
+    console.warn(`PayPal API order ${orderID}: capture failed`, data);
+  } else {
+    console.info(
+      `PayPal API order ${orderID}: successful capture`,
+      transaction
+    );
+    const capturedAmount = (<any>transaction?.amount)?.value;
```

Comment:
cast as paypal-js does not go into detail here https://github.com/paypal/paypal-js/blob/963aed995062eb72a1542b7b76a1e92cd3d9f732/types/apis/orders.d.ts#L183